### PR TITLE
task_DEV-7

### PR DESCRIPTION
**Pull request by shpashkeev**

### CODE REVIEW

File: DEV-7/DetectorTriangleType.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  class DetectorTriangleType
+  {
+    private double sideA;
+    private double sideB;
+    private double sideC;
+    const double EPSILON = 10e-6;
+
+    public TypeOfTriangles GetTriangleType(double[] sides)
+    {
+      sideA = sides[0];
```
Comment (IgorKheidorov) [inline DEV-7/DetectorTriangleType.cs]: Check for null reference

File: DEV-7/EquilateralorTriangle.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+{
+  class EquilateralorTriangle : Triangle
+  {
+    const string EQUILATERALOR = "Equilateralor";
+
+    public EquilateralorTriangle(double[] sides) : base(sides) { }
+
+    public override string Type()
+    {
+      return EQUILATERALOR;
```
Comment (IgorKheidorov) [inline DEV-7/EquilateralorTriangle.cs]: Return enum type

File: DEV-7/EquilateralorTriangle.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    }
```
Comment (IgorKheidorov) [inline DEV-7/EquilateralorTriangle.cs]: Add type output method as string

File: DEV-7/DetectorTriangleType.cs (hunk 1 above)
Comment (shpashkeev) [reply DEV-7/DetectorTriangleType.cs]: I create an exception "Error building triangle" in class TriangleBuilder

File: DEV-7/DetectorTriangleType.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  class DetectorTriangleType
+  {
+    private double sideA;
+    private double sideB;
+    private double sideC;
+    const double EPSILON = 10e-6;
+
+    public TypeOfTriangles GetTriangleType(double[] sides)
+    {
+
```
Comment (IgorKheidorov) [inline DEV-7/DetectorTriangleType.cs]: Extra line

File: DEV-7/DetectorTriangleType.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
+﻿using System;
+
+namespace TypeOfTriangle
+{
+  class DetectorTriangleType
```
Comment (IgorKheidorov) [inline DEV-7/DetectorTriangleType.cs]: Name

File: DEV-7/TriangleBuilder.cs [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+      {
+        TypeOfTriangles type = (new DetectorTriangleType()).GetTriangleType(values);
+        Triangle builtTriangle;
+        switch (type)
+        {
+          // When detected type is Not-Existent the triangle is constructed as Not-Existent
+          case TypeOfTriangles.NotExistent:
+            builtTriangle = new NotExistentTriangle(values);
+            break;
+
```
Comment (IgorKheidorov) [inline DEV-7/TriangleBuilder.cs]: Extra line

File: DEV-7/TriangleBuilder.cs [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    public Triangle Build(double[] values)
+    {
+      try
+      {
+        TypeOfTriangles type = (new DetectorTriangleType()).GetTriangleType(values);
+        Triangle builtTriangle;
+        switch (type)
+        {
+          // When detected type is Not-Existent the triangle is constructed as Not-Existent
```
Comment (IgorKheidorov) [inline DEV-7/TriangleBuilder.cs]: Remove comment

File: DEV-7/UsualTriangle.cs [hunk 8]
```diff
@@ -0,0 +1,16 @@
+﻿
+using System;
+
+namespace TypeOfTriangle
+{
+  class UsualTriangle : Triangle
+  {
+
```
Comment (IgorKheidorov) [inline DEV-7/UsualTriangle.cs]: Extra line

**Review by IgorKheidorov — APPROVED** [review]

**Pull request merged**